src/plugins/data_source/server/data_source_service.ts

```diff
@@ -3,15 +3,16 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { LegacyCallAPIOptions, Logger, OpenSearchClient } from '../../../../src/core/server';
+import { Client } from '@opensearch-project/opensearch';
+import { LegacyCallAPIOptions, Logger } from '../../../../src/core/server';
 import { DataSourcePluginConfigType } from '../config';
 import { configureClient, OpenSearchClientPool } from './client';
 import { configureLegacyClient } from './legacy';
 import { DataSourceClientParams } from './types';
 import { DataSourceAttributes } from '../common/data_sources';
 import { configureTestClient } from './client/configure_client';
 export interface DataSourceServiceSetup {
-  getDataSourceClient: (params: DataSourceClientParams) => Promise<OpenSearchClient>;
+  getDataSourceClient: (params: DataSourceClientParams) => Promise<Client>;
```

Comment:
why we need to change the client type here?

Reply:
reverted